module/functions/azure/Invoke-ArmTemplateDeployment.ps1

```diff
@@ -118,16 +123,21 @@ function Invoke-ArmTemplateDeployment
         $argsForDeployType += @{ Location = $Location }
     }
 
-    Write-Host "Validating ARM template ($ArmTemplatePath)..."
-    # Dynamically call the relevant cmdlet for the current deployment type
-    $validationErrors = & "Test-Az$($DeploymentScope)Deployment" `
-                                    @argsForDeployType `
-                                    @OptionalParameters `
-                                    @TemplateParameters `
-                                    -Verbose
-    if ($validationErrors) {
-        Write-Warning ($validationErrors | Out-String)
-        throw "ARM Template validation errors - check previous warnings"
+    if (!$SkipTestDeployment -and !$WhatIfPreference) {
```

Comment:
What's the reasoning behind also skipping validation when `-WhatIf` is set?

Reply:
The idea was that if nothing's going to get deployed then we can just let the `Invoke` command do its validation.